src/utils/src/Semaphore.c

```diff
@@ -385,6 +389,8 @@ STATUS semaphoreGetCountInternal(PSemaphore pSemaphore, PINT32 pCount)
     *pCount = (INT32) ATOMIC_LOAD(&pSemaphore->permitCount);
 
 CleanUp:
+    CHK_LOG_ERR(retStatus);
```

Comment:
This is going to spam a LOT of error logs for many instances where returning a non-success is completely benign / not an error at all.  IMO it makes much more sense for the client application or higher level data structure that is using to determine whether what's returned from here is an error and let it decide if it wants to bail out early and log the error.